Add label: AverTouch

duderin0 opened the pull request:
Results:

admin in ~/Documents/GitHub/Installomator on AverTouch-label λ sudo ./build/Installomator.sh avertouch DEBUG=0
Password:
2023-05-02 15:15:46 : INFO  : avertouch : setting variable from argument DEBUG=0
2023-05-02 15:15:46 : REQ   : avertouch : ################## Start Installomator v. 10.4beta, date 2023-05-02
2023-05-02 15:15:46 : INFO  : avertouch : ################## Version: 10.4beta
2023-05-02 15:15:46 : INFO  : avertouch : ################## Date: 2023-05-02
2023-05-02 15:15:46 : INFO  : avertouch : ################## avertouch
2023-05-02 15:15:47 : INFO  : avertouch : BLOCKING_PROCESS_ACTION=tell_user
2023-05-02 15:15:47 : INFO  : avertouch : NOTIFY=success
2023-05-02 15:15:47 : INFO  : avertouch : LOGGING=INFO
2023-05-02 15:15:47 : INFO  : avertouch : LOGO=/System/Applications/App Store.app/Contents/Resources/AppIcon.icns
2023-05-02 15:15:47 : INFO  : avertouch : Label type: appInDmgInZip
2023-05-02 15:15:47 : INFO  : avertouch : archiveName: AverTouch.zip
2023-05-02 15:15:47 : INFO  : avertouch : no blocking processes defined, using AverTouch as default
2023-05-02 15:15:47 : INFO  : avertouch : name: AverTouch, appName: AverTouch.app
2023-05-02 15:15:47.283 mdfind[22429:546864] [UserQueryParser] Loading keywords and predicates for locale "en_US"
2023-05-02 15:15:47.283 mdfind[22429:546864] [UserQueryParser] Loading keywords and predicates for locale "en"
2023-05-02 15:15:47.320 mdfind[22429:546864] Couldn't determine the mapping between prefab keywords and predicates.
2023-05-02 15:15:47 : WARN  : avertouch : No previous app found
2023-05-02 15:15:47 : WARN  : avertouch : could not find AverTouch.app
2023-05-02 15:15:47 : INFO  : avertouch : appversion: 
2023-05-02 15:15:47 : INFO  : avertouch : Latest version of AverTouch is 1.3.2084
2023-05-02 15:15:47 : REQ   : avertouch : Downloading https://www.averusa.com/education/downloads/AVerTouch_mac_v1.15(1.3.2084).zip to AverTouch.zip
2023-05-02 15:16:10 : REQ   : avertouch : no more blocking processes, continue with update
2023-05-02 15:16:10 : REQ   : avertouch : Installing AverTouch
2023-05-02 15:16:10 : INFO  : avertouch : Unzipping AverTouch.zip
2023-05-02 15:16:10 : INFO  : avertouch : found dmg: /var/folders/zz/zyxvpxvq6csfxvn_n0000000000000/T/tmp.gYCavZij/AVerTouch.dmg
2023-05-02 15:16:10 : INFO  : avertouch : Mounting /var/folders/zz/zyxvpxvq6csfxvn_n0000000000000/T/tmp.gYCavZij/AVerTouch.dmg
2023-05-02 15:16:13 : INFO  : avertouch : Mounted: /Volumes/AVerTouch
2023-05-02 15:16:13 : INFO  : avertouch : Verifying: /Volumes/AVerTouch/AverTouch.app
2023-05-02 15:16:15 : INFO  : avertouch : Team ID matching: B6T3WCD59Q (expected: B6T3WCD59Q )
2023-05-02 15:16:15 : INFO  : avertouch : Installing AverTouch version 1.3.2084 on versionKey CFBundleVersion.
2023-05-02 15:16:15 : INFO  : avertouch : App has LSMinimumSystemVersion: 10.13
2023-05-02 15:16:15 : INFO  : avertouch : Copy /Volumes/AVerTouch/AverTouch.app to /Applications
2023-05-02 15:16:16 : WARN  : avertouch : Changing owner to admin
2023-05-02 15:16:16 : INFO  : avertouch : Finishing...
2023-05-02 15:16:19 : INFO  : avertouch : App(s) found: /Applications/AverTouch.app
2023-05-02 15:16:19 : INFO  : avertouch : found app at /Applications/AverTouch.app, version 1.3.2084, on versionKey CFBundleVersion
2023-05-02 15:16:19 : REQ   : avertouch : Installed AverTouch, version 1.3.2084
2023-05-02 15:16:19 : INFO  : avertouch : notifying
2023-05-02 15:16:20 : INFO  : avertouch : App not closed, so no reopen.
2023-05-02 15:16:20 : REQ   : avertouch : All done!
2023-05-02 15:16:20 : REQ   : avertouch : ################## End Installomator, exit code 0

scriptingosx commented:
getting an error here. the file it downloads is actually an html page (probably an error message)

duderin0 commented:
Copying from commit:

Fixing everything after site update
Aver seems to be using two different version numbers, 1.15 (reflected in CFBundleShortVersionString) and 1.3.2087 (is CFBundleVersion and is the one visible to users and on the site). They previously used  the 1.15 in the filename, but they've removed that part now.

Changed grep and sed to xmllint instead, to be a little more understandable than the previous sed command. Should help if they decide to do odd naming/versions again, but I'm hoping to contact them and get something consistent going forward.

They also changed to just an .app in a .zip. Hoping to get that consistent as well.

New output:
./assemble.sh avertouch
2023-05-30 11:15:33 : REQ   : avertouch : ################## Start Installomator v. 10.4beta, date 2023-05-30
2023-05-30 11:15:33 : INFO  : avertouch : ################## Version: 10.4beta
2023-05-30 11:15:33 : INFO  : avertouch : ################## Date: 2023-05-30
2023-05-30 11:15:33 : INFO  : avertouch : ################## avertouch
2023-05-30 11:15:33 : DEBUG : avertouch : DEBUG mode 1 enabled.
2023-05-30 11:15:33 : DEBUG : avertouch : name=AverTouch
2023-05-30 11:15:33 : DEBUG : avertouch : appName=
2023-05-30 11:15:33 : DEBUG : avertouch : type=zip
2023-05-30 11:15:33 : DEBUG : avertouch : archiveName=
2023-05-30 11:15:33 : DEBUG : avertouch : downloadURL=https://www.averusa.com/education/downloads/AVerTouch_mac_v1.3.2087.zip
2023-05-30 11:15:33 : DEBUG : avertouch : curlOptions=
2023-05-30 11:15:33 : DEBUG : avertouch : appNewVersion=1.3.2087
2023-05-30 11:15:33 : DEBUG : avertouch : appCustomVersion function: Not defined
2023-05-30 11:15:33 : DEBUG : avertouch : versionKey=CFBundleVersion
2023-05-30 11:15:33 : DEBUG : avertouch : packageID=
2023-05-30 11:15:33 : DEBUG : avertouch : pkgName=
2023-05-30 11:15:33 : DEBUG : avertouch : choiceChangesXML=
2023-05-30 11:15:33 : DEBUG : avertouch : expectedTeamID=B6T3WCD59Q
2023-05-30 11:15:33 : DEBUG : avertouch : blockingProcesses=
2023-05-30 11:15:33 : DEBUG : avertouch : installerTool=
2023-05-30 11:15:33 : DEBUG : avertouch : CLIInstaller=
2023-05-30 11:15:33 : DEBUG : avertouch : CLIArguments=
2023-05-30 11:15:33 : DEBUG : avertouch : updateTool=
2023-05-30 11:15:33 : DEBUG : avertouch : updateToolArguments=
2023-05-30 11:15:33 : DEBUG : avertouch : updateToolRunAsCurrentUser=
2023-05-30 11:15:33 : INFO  : avertouch : BLOCKING_PROCESS_ACTION=tell_user
2023-05-30 11:15:33 : INFO  : avertouch : NOTIFY=success
2023-05-30 11:15:33 : INFO  : avertouch : LOGGING=DEBUG
2023-05-30 11:15:33 : INFO  : avertouch : LOGO=/System/Applications/App Store.app/Contents/Resources/AppIcon.icns
2023-05-30 11:15:33 : INFO  : avertouch : Label type: zip
2023-05-30 11:15:33 : INFO  : avertouch : archiveName: AverTouch.zip
2023-05-30 11:15:33 : INFO  : avertouch : no blocking processes defined, using AverTouch as default
2023-05-30 11:15:33 : DEBUG : avertouch : Changing directory to /Users/admin/Documents/GitHub/Installomator/build
2023-05-30 11:15:33 : INFO  : avertouch : App(s) found: /Applications/AverTouch.app
2023-05-30 11:15:34 : INFO  : avertouch : found app at /Applications/AverTouch.app, version 1.3.2084, on versionKey CFBundleVersion
2023-05-30 11:15:34 : INFO  : avertouch : appversion: 1.3.2084
2023-05-30 11:15:34 : INFO  : avertouch : Latest version of AverTouch is 1.3.2087
2023-05-30 11:15:34 : REQ   : avertouch : Downloading https://www.averusa.com/education/downloads/AVerTouch_mac_v1.3.2087.zip to AverTouch.zip
2023-05-30 11:15:34 : DEBUG : avertouch : No Dialog connection, just download
2023-05-30 11:15:48 : DEBUG : avertouch : File list: -rw-r--r--@ 1 admin  2103187081   148M May 30 11:15 AverTouch.zip
2023-05-30 11:15:48 : DEBUG : avertouch : File type: AverTouch.zip: Zip archive data, at least v2.0 to extract, compression method=store
2023-05-30 11:15:48 : DEBUG : avertouch : curl output was:
*   Trying 54.153.15.164:443...
* Connected to [www.averusa.com](http://www.averusa.com/) (54.153.15.164) port 443 (#0)
* ALPN: offers h2,http/1.1
* (304) (OUT), TLS handshake, Client hello (1):
} [320 bytes data]
*  CAfile: /etc/ssl/cert.pem
*  CApath: none
* (304) (IN), TLS handshake, Server hello (2):
{ [81 bytes data]
* TLSv1.2 (IN), TLS handshake, Certificate (11):
{ [3006 bytes data]
* TLSv1.2 (IN), TLS handshake, Server key exchange (12):
{ [401 bytes data]
* TLSv1.2 (IN), TLS handshake, Server finished (14):
{ [4 bytes data]
* TLSv1.2 (OUT), TLS handshake, Client key exchange (16):
} [138 bytes data]
* TLSv1.2 (OUT), TLS change cipher, Change cipher spec (1):
} [1 bytes data]
* TLSv1.2 (OUT), TLS handshake, Finished (20):
} [16 bytes data]
* TLSv1.2 (IN), TLS change cipher, Change cipher spec (1):
{ [1 bytes data]
* TLSv1.2 (IN), TLS handshake, Finished (20):
{ [16 bytes data]
* SSL connection using TLSv1.2 / ECDHE-RSA-AES256-SHA384
* ALPN: server did not agree on a protocol. Uses default.
* Server certificate:
*  subject: C=US; ST=California; L=Fremont; O=AVer Information Inc.; CN=averusa.com
*  start date: Jan 27 00:00:00 2023 GMT
*  expire date: Feb 27 23:59:59 2024 GMT
*  subjectAltName: host "[www.averusa.com](http://www.averusa.com/)" matched cert's "[www.averusa.com](http://www.averusa.com/)"
*  issuer: C=US; O=DigiCert Inc; CN=DigiCert TLS RSA SHA256 2020 CA1
*  SSL certificate verify ok.
* using HTTP/1.x
> GET /education/downloads/AVerTouch_mac_v1.3.2087.zip HTTP/1.1
> Host: [www.averusa.com](http://www.averusa.com/)
> User-Agent: curl/7.88.1
> Accept: */*
>
< HTTP/1.1 200 OK
< Content-Type: application/x-zip-compressed
< Last-Modified: Thu, 25 May 2023 18:21:16 GMT
< Accept-Ranges: bytes
< ETag: "0769cb1358fd91:0"
< Server: Microsoft-IIS/7.5
< X-Powered-By: ASP.NET
< Date: Tue, 30 May 2023 18:43:29 GMT
< Content-Length: 155270375
<
{ [16113 bytes data]
* Connection #0 to host [www.averusa.com](http://www.averusa.com/) left intact

2023-05-30 11:15:48 : DEBUG : avertouch : DEBUG mode 1, not checking for blocking processes
2023-05-30 11:15:48 : REQ   : avertouch : Installing AverTouch
2023-05-30 11:15:48 : INFO  : avertouch : Unzipping AverTouch.zip
2023-05-30 11:15:49 : INFO  : avertouch : Verifying: /Users/admin/Documents/GitHub/Installomator/build/AverTouch.app
2023-05-30 11:15:49 : DEBUG : avertouch : App size: 257M	/Users/admin/Documents/GitHub/Installomator/build/AverTouch.app
2023-05-30 11:15:49 : DEBUG : avertouch : Debugging enabled, App Verification output was:
/Users/admin/Documents/GitHub/Installomator/build/AverTouch.app: accepted
source=Notarized Developer ID
origin=Developer ID Application: AVer Information Inc. (B6T3WCD59Q)

2023-05-30 11:15:49 : INFO  : avertouch : Team ID matching: B6T3WCD59Q (expected: B6T3WCD59Q )
2023-05-30 11:15:50 : INFO  : avertouch : Downloaded version of AverTouch is 1.3.2087 on versionKey CFBundleVersion (replacing version 1.3.2084).
2023-05-30 11:15:50 : INFO  : avertouch : App has LSMinimumSystemVersion: 10.15
2023-05-30 11:15:50 : DEBUG : avertouch : DEBUG mode 1 enabled, skipping remove, copy and chown steps
2023-05-30 11:15:50 : INFO  : avertouch : Finishing...
2023-05-30 11:15:53 : INFO  : avertouch : App(s) found: /Applications/AverTouch.app
2023-05-30 11:15:53 : INFO  : avertouch : found app at /Applications/AverTouch.app, version 1.3.2084, on versionKey CFBundleVersion
2023-05-30 11:15:53 : REQ   : avertouch : Installed AverTouch, version 1.3.2084
2023-05-30 11:15:53 : INFO  : avertouch : notifying
2023-05-30 11:15:53 : DEBUG : avertouch : DEBUG mode 1, not reopening anything
2023-05-30 11:15:53 : REQ   : avertouch : All done!
2023-05-30 11:15:53 : REQ   : avertouch : ################## End Installomator, exit code 0

scriptingosx commented:
thank you!
